docs/dev/RELEASE.md

```diff
@@ -0,0 +1,87 @@
+## Introduction
+This document guides maintainers through the process of creating an Envoy Gateway release.
+
+## Prerequisites
+- Permissions to push to the Envoy Gateway repository.
+
+## Creating a Minor Release
+
+1. Clone the repo, checkout the `main` branch, ensure it’s up-to-date, and your local branch is clean.
+2. Create a changelog corresponding to the release number that summarizes the changes included in the release.
+   Reference previous [changelogs][changelogs] for additional details.
+3. Submit a [Pull Request][pr] to merge the changelog into the main branch. This should be the last commit to main
+   before cutting the release.
+4. Create a new release branch from `main`. The release branch should be named
+   `release/v${MAJOR_VERSION}.${MINOR_VERSION}.0`, e.g. `release/v0.3.0`.
+   ```shell
+   git checkout -b release/v0.3.0
+   ```
+5. Push the branch to the Envoy Gateway repo.
+6. Create a topic branch and update the release tag references in the [Quickstart Guide][quickstart].
+7. Sign, commit, and push your changes to your fork. Send a PR to get your changes merged to the release branch.
+   Do not proceed until your PR is merged into the release branch.
+8. Confirm that the [release workflow][release_workflow] for your PR completed successfully.
+9. Tag the head of your release branch with the release tag. For example:
+   ```shell
+   git tag -a v0.3.0 -m 'Envoy Gateway v0.3.0 Release'
+   ```
+10. Push the tag to the Envoy Gateway repository.
+    ```shell
+    git push --tags
+    ```
+11. This will trigger the [release GitHub action][release_gha] that generates the release, release artifacts, etc.
+12. Confirm that the [release workflow][release_workflow] completed successfully.
+13. Confirm that the [Envoy Gateway image][image] with the correct release tag was published to Docker Hub.
+14. Confirm that the [release][release] was created.
+15. Confirm that the steps in the [Quickstart Guide][quickstart] work as expected.
+16. [Generate][release_notes] the release notes.
+17. Submit a PR to merge the Quickstart Guide changes from the release branch into the main branch.
+18. If you find any bugs in this process, please create an issue.
+
+## Creating a Release Candidate
```

Comment:
can this be combined with the above section with a note for line 4, saying if Release candidate add the `rc.X` suffix to the tag ?

Reply:
I did have both workflows initially combined, but I thought it started to get a bit confusing as they deviated.